clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/config/BootstrapKubernetesConfigurationProvider.java

```diff
@@ -0,0 +1,211 @@
+
+    // loop through each account and bind
+    for (Map<String, Object> unflattendAcc :
+        ((List<Map<String, Object>>) propertiesMap.get("accounts"))) {
+      result =
+          bind(getFlatMap(unflattendAcc), KubernetesConfigurationProperties.ManagedAccount.class);
+      k8sConfigProps
+          .getAccounts()
+          .add((KubernetesConfigurationProperties.ManagedAccount) result.get());
+    }
+
+    try {
+      propertiesMap.remove("accounts"); // accounts are already processed above
+      result = bind(getFlatMap(propertiesMap), KubernetesConfigurationProperties.class);
+      k8sConfigProps.setRawResourcesEndpointConfig(
+          ((KubernetesConfigurationProperties) result.get()).getRawResourcesEndpointConfig());
+      // mappings to any future fields similar to rawResourcesEndpointConfig go here
+    } catch (NoSuchElementException e) {
+      // this binding error occurs when rawResourcesEndpointConfig is not supplied.
+      // ignoring it as rawResourcesEndpointConfig is not a mandatory configuration
+    }
+    return k8sConfigProps;
+  }
+
+  private BindResult<?> bind(Map<String, Object> propertiesMap, Class<?> clazz) {
+    resolveSpecialCases(propertiesMap);
+    ConfigurationPropertySource configurationPropertySource =
+        new MapConfigurationPropertySource(propertiesMap);
+    Iterable<ConfigurationPropertySource> sourceIterable =
+        () -> Collections.singleton(configurationPropertySource).iterator();
+    Binder binder =
+        new Binder(
+            sourceIterable,
+            new PropertySourcesPlaceholdersResolver(applicationContext.getEnvironment()));
+    return binder.bind("", Bindable.of(clazz));
+  }
+
+  private void resolveSpecialCases(Map<String, Object> propertiesMap) {
+    String result;
+    for (Map.Entry<String, Object> entry : propertiesMap.entrySet()) {
+      if (entry.getValue() instanceof String) {
+        result = resolveConfigServerPattern((String) entry.getValue());
+        result = resolveEncryptedPattern(result);
+        entry.setValue(result);
+      }
+    }
+  }
+
+  private Map<String, Object> getFlatMap(Map<String, Object> unflatMap) {
+    try {
+      return JsonFlattener.flattenAsMap(objectMapper.writeValueAsString(unflatMap));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(
+          "Error occurred while building KubernetesConfigurationProperties object: "
+              + e.getMessage());
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private Map<String, Object> getPropertiesMap() {
+    ConfigurableEnvironment environment = applicationContext.getEnvironment();
+    Map<String, Object> map;
+
+    for (PropertySource<?> propertySource : environment.getPropertySources()) {
+
+      if (propertySource instanceof BootstrapPropertySource) {
+        map = (Map<String, Object>) propertySource.getSource();
+        if (map.containsKey(FIRST_ACCOUNT_NAME_KEY)) {
```

Comment:
The code returns on the first property source that has defined an account, instead of looking through the priority of `clouddriver-local.yml`, `clouddriver.yml`, `spinnaker-local.yml`, `spinnaker.yml`. How do you know that the right file is parsed first?

There may be existing spinnaker installations having accounts in `.hal/config` and accounts in `default/profiles/clouddriver-local.yml`, for which this code seems to have 50% chances of picking the right set of accounts.